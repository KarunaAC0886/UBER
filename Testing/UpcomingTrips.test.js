import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UpcomingTrips from '../myscreens/UpcomingTrips';

jest.mock('@react-navigation/material-top-tabs', () => {
  const Tab = jest.fn();
  const createMaterialTopTabNavigator = jest.fn(() => ({ Tab }));
  return { createMaterialTopTabNavigator };
});

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));

describe('Past Trips', () => {
  it('Past Trips page renders correctly', () => {
    const tree = renderer.create(<UpcomingTrips />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
