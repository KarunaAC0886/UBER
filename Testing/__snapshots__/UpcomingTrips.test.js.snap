// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Past Trips Past Trips page renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#1A1A1A",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": "80%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      Uber will plan to announce new trips
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ffffff",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 18,
          }
        }
      >
        Contact
      </Text>
    </View>
  </View>
</View>
`;
